File: LeetCode/kotlin/maxProduct.kt
import java.lang.StrictMath.max

fun maxProduct(nums: IntArray): Int {
    var largestProduct = nums[0]
    for (i in nums.indices) {
        var currentProduct = nums[i]
        for (j in (i + 1)..(nums.size - 1)) {
            largestProduct = max(largestProduct, currentProduct)
            currentProduct *= nums[j]
        }
        largestProduct = max(largestProduct, currentProduct)
    }
    return largestProduct
}

File: LeetCode/javascript/maxProduct.js
var maxProduct = function (nums) {
  let largestProduct = nums[0];
  for (let i = 0; i < nums.length; i++) {
    let currentProduct = nums[i];
    for (let j = i + 1; j < nums.length; j++) {
      largestProduct = Math.max(largestProduct, currentProduct);
      currentProduct = currentProduct * nums[j];
    }
    largestProduct = Math.max(largestProduct, currentProduct);
  }
  return largestProduct;
};
